src/Jackett.Common/Indexers/CardigannIndexer.cs

```diff
@@ -853,12 +854,19 @@ protected bool CheckIfLoginIsNeeded(WebResult Result, IHtmlDocument document)
             if (Definition.Login == null || Definition.Login.Test == null)
                 return false;
 
-            if (Definition.Login.Test.Selector != null)
+            // Only run html test selector on html responses
+            if (Result.Headers.ContentType.Contains("text/html"))
```

Comment:
Might need .Headers["Content-Type"] here, or however you guys would normally grab it